lib/matobj/positional.gi

```diff
@@ -0,0 +1,88 @@
+#############################################################################
+##
+##  This file is part of GAP, a system for computational discrete algebra.
+##
+##  SPDX-License-Identifier: GPL-2.0-or-later
+##
+##  Copyright of GAP belongs to its developers, whose names are too numerous
+##  to list here. Please refer to the COPYRIGHT file for details.
+##
+
+############################################################################
+#
+# Operations for positional matrix objects
+#
+############################################################################
+
+InstallMethod( BaseDomain, [ IsPositionalVectorRep ],
+  function( v )
+    return v![VEC_BD_POS];
+  end );
+
+InstallMethod( Length, [ IsPositionalVectorRep ],
+  function( v )
+    return Length(v![VEC_DATA_POS]); # FIXME: assumptions
+  end );
+
+
+InstallMethod( ShallowCopy, [ IsPositionalVectorRep ],
+  function( v )
+    local i, res;
+    res := List([1..LEN_POSOBJ(v)], i -> v![i]);
+    res![VEC_DATA_POS] := ShallowCopy(v![VEC_DATA_POS]);
+    res := Objectify(TypeObj(v), res);
+
+    # 'ShallowCopy' MUST return a mutable object if such an object exists at all!
```

Comment:
My interpretation of `ShallowCopy` is that it returns its argument if this is not copyable.
We could restrict the method installation to `IsPositionalVectorRep and IsCopyable`.
And perhaps we should add an explicit reference to `IsCopyable` to the documentation of `ShallowCopy`.